ports/nrf/drivers/bluetooth/ble_uart.c

```diff
@@ -110,9 +110,10 @@ int mp_hal_stdin_rx_chr(void) {
     return (int)byte;
 }
 
-void mp_hal_stdout_tx_strn(const char *str, size_t len) {
+mp_uint_t mp_hal_stdout_tx_strn(const char *str, size_t len) {
+    mp_uint_t ret = len;
     // Not connected: drop output
-    if (!ble_uart_enabled()) return;
+    if (!ble_uart_enabled()) return ret;
```

Comment:
Suggest returning 0 here, same reasoning as above.

Reply:
Agree. In line with this, `if (MP_STATE_PORT(board_stdio_uart) == NULL)` in `nrf/mphalport.c` now also returns 0.

Reply:
Well spotted, thanks!